server/src/main/java/org/opensearch/search/query/TopDocsCollectorContext.java

```diff
@@ -595,8 +605,14 @@ TopDocsAndMaxScore newTopDocs(final TopDocs topDocs, final float maxScore, final
                 newTopDocs = new TopDocs(totalHits, scoreDocs);
             }
 
-            if (Float.isNaN(maxScore) && newTopDocs.scoreDocs.length > 0 && sortAndFormats == null) {
-                return new TopDocsAndMaxScore(newTopDocs, newTopDocs.scoreDocs[0].score);
```

Comment:
This change is for the case that concurrent segment search is enabled because the logic is different from the default case.